library/Hasql/Pool.hs

```diff
@@ -72,9 +80,16 @@ acquire ::
   DiffTime ->
   -- | Connection settings.
   Connection.Settings ->
+  -- | Observation handler.
+  --
+  -- Typically it's used for monitoring the state of the pool via metrics and logging.
+  --
+  -- If the action is not lightweight, it's recommended to use intermediate bufferring via channels like TBQueue.
+  -- E.g., if the action is @'atomically' . 'writeTBQueue' yourQueue@, then reading from it and processing can be done on a separate thread.
```

Comment:
When could this happen? When the pool size is big and there are lots of connections?

Reply:
I'm talking about the user supplied action here. It should be quick since it will block the pool processes.